src/test/fuzz/util.cpp

```diff
@@ -7,6 +7,14 @@
 #include <util/rbf.h>
 #include <version.h>
 
+bool FuzzedSock::Wait(std::chrono::milliseconds timeout, Event requested, Event* occurred ) const
+{
+    if (!m_fuzzed_data_provider.ConsumeBool()) {
+        return false;
+    }
+    if (occurred) *occurred = 0;
+    return true;
+}
```

Comment:
* This would signal a failure without setting `errno` (it was like this even before this PR, but better fix it)
* Would result in either an error or timeout, but never "the requested event occurred"
* `!ConsumeBool()` is the same as `ConsumeBool()`

```suggestion
{
    constexpr std::array wait_errnos{
        EBADF,
        EINTR,
        EINVAL,
    };
    if (m_fuzzed_data_provider.ConsumeBool()) { 
        SetFuzzedErrNo(m_fuzzed_data_provider, wait_errnos);
        return false;
    } 
    if (occurred) { 
        if (m_fuzzed_data_provider.ConsumeBool()) { 
            *occurred = requested;
        } else { 
            *occurred = 0;
        } 
    } 
    return true;
}
```

Reply:
> !ConsumeBool() is the same as ConsumeBool()

I had a slight preference to not invalidate the existing fuzz inputs by using the `!` operator on the bool

Mind taking the other changes to a new pr?